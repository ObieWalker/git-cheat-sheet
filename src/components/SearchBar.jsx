import React from 'react';
import '../styles/App.css';

const SearchBar = ({updateSearch}) => {
  return (
    <div className="container">
      <br/>
      <div className="row justify-content-center" style={{ width: '50%', float: 'none', margin: 'auto 33%' }}>
        <div className="col-12 col-md-10 col-lg-8">
            <form className="card card-sm">
                <div className="card-body row no-gutters align-items-center">
                    <div className="col-auto">
                    </div>
                    <div className="col">
                        <input className="form-control form-control-lg form-control-borderless fa"
                        
                        type="search"
                        name='search'
                        onChange={updateSearch.bind(this)}
                        placeholder="&#61442; THE AWESOME GIT CHEAT SHEET"
                        />
                    </div>
                </div>
            </form>
        </div>
      </div>
    </div>
    );
  }

export default SearchBar;
