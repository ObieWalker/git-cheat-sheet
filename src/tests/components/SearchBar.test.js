import React from 'react';
import ReactDOM from 'react-dom';
import expect from 'expect';
import { shallow } from 'enzyme';
import SearchBar from '../../components/SearchBar';

let mounted;
let props;
let wrapper;
const updateSearch = jest.fn(() => Promise.resolve({}));

const getComponent = () => {
  if (!mounted) {
    props = {
      updateSearch
    };
    mounted = shallow(<SearchBar {...props} />);
  }
  return mounted;
};

describe('cheats Component', () => {
  beforeEach(() => {});

  it('renders component successfully', () => {
    wrapper = getComponent();
    expect(wrapper).toMatchSnapshot();
  });

  it('should click on delete cheat button', () => {
    wrapper = getComponent();
    wrapper
    .find('.form-control')
    .simulate('change', { target: { name: 'search', value: 'obi' } });
  });
});

