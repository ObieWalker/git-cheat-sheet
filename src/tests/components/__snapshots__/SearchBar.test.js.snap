// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`cheats Component renders component successfully 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SearchBar
    updateSearch={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <br />,
        <div
          className="row justify-content-center"
          style={
            Object {
              "float": "none",
              "margin": "auto 33%",
              "width": "50%",
            }
          }
        >
          <div
            className="col-12 col-md-10 col-lg-8"
          >
            <form
              className="card card-sm"
            >
              <div
                className="card-body row no-gutters align-items-center"
              >
                <div
                  className="col-auto"
                />
                <div
                  className="col"
                >
                  <input
                    className="form-control form-control-lg form-control-borderless fa"
                    name="search"
                    onChange={[Function]}
                    placeholder=" THE AWESOME GIT CHEAT SHEET"
                    type="search"
                  />
                </div>
              </div>
            </form>
          </div>
        </div>,
      ],
      "className": "container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": "br",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="col-12 col-md-10 col-lg-8"
          >
            <form
              className="card card-sm"
            >
              <div
                className="card-body row no-gutters align-items-center"
              >
                <div
                  className="col-auto"
                />
                <div
                  className="col"
                >
                  <input
                    className="form-control form-control-lg form-control-borderless fa"
                    name="search"
                    onChange={[Function]}
                    placeholder=" THE AWESOME GIT CHEAT SHEET"
                    type="search"
                  />
                </div>
              </div>
            </form>
          </div>,
          "className": "row justify-content-center",
          "style": Object {
            "float": "none",
            "margin": "auto 33%",
            "width": "50%",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <form
              className="card card-sm"
            >
              <div
                className="card-body row no-gutters align-items-center"
              >
                <div
                  className="col-auto"
                />
                <div
                  className="col"
                >
                  <input
                    className="form-control form-control-lg form-control-borderless fa"
                    name="search"
                    onChange={[Function]}
                    placeholder=" THE AWESOME GIT CHEAT SHEET"
                    type="search"
                  />
                </div>
              </div>
            </form>,
            "className": "col-12 col-md-10 col-lg-8",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <div
                className="card-body row no-gutters align-items-center"
              >
                <div
                  className="col-auto"
                />
                <div
                  className="col"
                >
                  <input
                    className="form-control form-control-lg form-control-borderless fa"
                    name="search"
                    onChange={[Function]}
                    placeholder=" THE AWESOME GIT CHEAT SHEET"
                    type="search"
                  />
                </div>
              </div>,
              "className": "card card-sm",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <div
                    className="col-auto"
                  />,
                  <div
                    className="col"
                  >
                    <input
                      className="form-control form-control-lg form-control-borderless fa"
                      name="search"
                      onChange={[Function]}
                      placeholder=" THE AWESOME GIT CHEAT SHEET"
                      type="search"
                    />
                  </div>,
                ],
                "className": "card-body row no-gutters align-items-center",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "className": "col-auto",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": "div",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": <input
                      className="form-control form-control-lg form-control-borderless fa"
                      name="search"
                      onChange={[Function]}
                      placeholder=" THE AWESOME GIT CHEAT SHEET"
                      type="search"
                    />,
                    "className": "col",
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "form-control form-control-lg form-control-borderless fa",
                      "name": "search",
                      "onChange": [Function],
                      "placeholder": " THE AWESOME GIT CHEAT SHEET",
                      "type": "search",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "input",
                  },
                  "type": "div",
                },
              ],
              "type": "div",
            },
            "type": "form",
          },
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <br />,
          <div
            className="row justify-content-center"
            style={
              Object {
                "float": "none",
                "margin": "auto 33%",
                "width": "50%",
              }
            }
          >
            <div
              className="col-12 col-md-10 col-lg-8"
            >
              <form
                className="card card-sm"
              >
                <div
                  className="card-body row no-gutters align-items-center"
                >
                  <div
                    className="col-auto"
                  />
                  <div
                    className="col"
                  >
                    <input
                      className="form-control form-control-lg form-control-borderless fa"
                      name="search"
                      onChange={[Function]}
                      placeholder=" THE AWESOME GIT CHEAT SHEET"
                      type="search"
                    />
                  </div>
                </div>
              </form>
            </div>
          </div>,
        ],
        "className": "container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "br",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="col-12 col-md-10 col-lg-8"
            >
              <form
                className="card card-sm"
              >
                <div
                  className="card-body row no-gutters align-items-center"
                >
                  <div
                    className="col-auto"
                  />
                  <div
                    className="col"
                  >
                    <input
                      className="form-control form-control-lg form-control-borderless fa"
                      name="search"
                      onChange={[Function]}
                      placeholder=" THE AWESOME GIT CHEAT SHEET"
                      type="search"
                    />
                  </div>
                </div>
              </form>
            </div>,
            "className": "row justify-content-center",
            "style": Object {
              "float": "none",
              "margin": "auto 33%",
              "width": "50%",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <form
                className="card card-sm"
              >
                <div
                  className="card-body row no-gutters align-items-center"
                >
                  <div
                    className="col-auto"
                  />
                  <div
                    className="col"
                  >
                    <input
                      className="form-control form-control-lg form-control-borderless fa"
                      name="search"
                      onChange={[Function]}
                      placeholder=" THE AWESOME GIT CHEAT SHEET"
                      type="search"
                    />
                  </div>
                </div>
              </form>,
              "className": "col-12 col-md-10 col-lg-8",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <div
                  className="card-body row no-gutters align-items-center"
                >
                  <div
                    className="col-auto"
                  />
                  <div
                    className="col"
                  >
                    <input
                      className="form-control form-control-lg form-control-borderless fa"
                      name="search"
                      onChange={[Function]}
                      placeholder=" THE AWESOME GIT CHEAT SHEET"
                      type="search"
                    />
                  </div>
                </div>,
                "className": "card card-sm",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <div
                      className="col-auto"
                    />,
                    <div
                      className="col"
                    >
                      <input
                        className="form-control form-control-lg form-control-borderless fa"
                        name="search"
                        onChange={[Function]}
                        placeholder=" THE AWESOME GIT CHEAT SHEET"
                        type="search"
                      />
                    </div>,
                  ],
                  "className": "card-body row no-gutters align-items-center",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "col-auto",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "div",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": <input
                        className="form-control form-control-lg form-control-borderless fa"
                        name="search"
                        onChange={[Function]}
                        placeholder=" THE AWESOME GIT CHEAT SHEET"
                        type="search"
                      />,
                      "className": "col",
                    },
                    "ref": null,
                    "rendered": Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "form-control form-control-lg form-control-borderless fa",
                        "name": "search",
                        "onChange": [Function],
                        "placeholder": " THE AWESOME GIT CHEAT SHEET",
                        "type": "search",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "input",
                    },
                    "type": "div",
                  },
                ],
                "type": "div",
              },
              "type": "form",
            },
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
